### PR TITLE
Update mithril.d.ts

### PR DESCRIPTION
**Pull request by sebcoetzee**
Updates mithril typing definitions. Adds some API functionality that was not there previously.

### CODE REVIEW

**Comment by vvakame** [comment]
*mithril/mithril.d.ts*

to authors(@lhorie @cbowdon). could you review this PR?
:+1: or :-1:?

check list

* [ ] pass the Travic-CI test?

File: mithril/mithril.d.ts [hunk 1]
```diff
@@ -28,6 +24,22 @@ interface MithrilStatic {
...
 
+interface MithrilRoute {
+	(rootElement: Element, defaultRoute: string, routes: { [key: string]: MithrilModule }): void;
+	(rootElement: HTMLDocument, defaultRoute: string, routes: { [key: string]: MithrilModule }): void;
+	(path: string, params?: any, shouldReplaceHistory?: boolean): void;
+	(element: Element, isInitialized: boolean): void;
+	(): string;
+	mode: string;
+	param: MithrilParam;
+	buildQueryString(data: Object): string;
```
Comment (cbowdon) [inline mithril/mithril.d.ts]: It would be nice to have a more descriptive type than Object here, if possible.

**Comment by cbowdon** [comment]
This looks good to include. That said, @SebastianCoetzee if you are still available to make some changes it would be nice to have something more descriptive than Object to define the new methods.

**Comment by vvakame** [comment]
@cbowdon thanks!
@SebastianCoetzee thanks mate!

**Pull request merged**